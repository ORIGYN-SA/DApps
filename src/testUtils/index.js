import React from 'react';
import TestEnvironment from './testEnv'

export * from '@testing-library/react';

export const makeFormikMethods = (mockFn) => ({
  resetForm: mockFn,
  setErrors: mockFn,
  setFieldError: mockFn,
  setFieldTouched: mockFn,
  setFieldValue: mockFn,
  setFormikState: mockFn,
  setStatus: mockFn,
  setSubmitting: mockFn,
  setTouched: mockFn,
  setValues: mockFn,
  submitForm: mockFn,
  validateField: mockFn,
  validateForm: mockFn,
});

module.exports = TestEnvironment