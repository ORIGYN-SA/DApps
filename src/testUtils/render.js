import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import { createMemoryHistory } from 'history';

function customRender(children) {
    const history = createMemoryHistory({ initialEntries: ['/'] });
    return render(<Router history={history}>{children}</Router>);
}

export { customRender as render };