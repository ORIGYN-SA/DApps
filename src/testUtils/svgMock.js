import React from 'react';

const SvgMock = () => <svg></svg>;

export default SvgMock;
