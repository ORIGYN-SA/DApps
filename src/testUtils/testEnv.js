import Environment from 'jest-environment-jsdom-global'
class TestEnvironment extends Environment {
  constructor(config, context) {
    super(config, context);
    if (typeof this.global.TextEncoder === 'undefined') {
      const { TextEncoder } = require('util');
      this.global.TextEncoder = TextEncoder;
    }
    if (typeof this.global.TextDecoder === 'undefined') {
      const { TextDecoder } = require('util');
      this.global.TextDecoder = TextDecoder;
    }
    if (typeof this.global.fetch === 'undefined') {
      const fetch = require('isomorphic-fetch');
      this.global.fetch = fetch;
    }
  }
};

module.exports = TestEnvironment