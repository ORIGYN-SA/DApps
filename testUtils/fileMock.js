import module from 'module';
module.exports = 'test-file-stub';
