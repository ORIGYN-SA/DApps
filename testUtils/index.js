import React from 'react'
import { render } from '@testing-library/react'
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom'

import { createMemoryHistory } from 'history'

function customRender(children) {
  function Wrapper({ children }) {
    const history = createMemoryHistory({ initialEntries: ['/'] })
    return <Router history={history}>{children}</Router>
  }
  return render(<Wrapper>{children}</Wrapper>)
}
export * from '@testing-library/react'

export { customRender as render }

export const makeFormikMethods = (mockFn) => ({
  resetForm: mockFn,
  setErrors: mockFn,
  setFieldError: mockFn,
  setFieldTouched: mockFn,
  setFieldValue: mockFn,
  setFormikState: mockFn,
  setStatus: mockFn,
  setSubmitting: mockFn,
  setTouched: mockFn,
  setValues: mockFn,
  submitForm: mockFn,
  validateField: mockFn,
  validateForm: mockFn,
})
