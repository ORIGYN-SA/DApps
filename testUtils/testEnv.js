const Environment = require('jest-environment-jsdom-global');

module.exports = class CustomTestEnvironment extends Environment {
  constructor({ globalConfig, projectConfig }, context) {
    super({ globalConfig, projectConfig }, context);
    if (typeof this.global.TextEncoder === 'undefined') {
      const { TextEncoder } = require('util');
      this.global.TextEncoder = TextEncoder;
    }
    if (typeof this.global.fetch === 'undefined') {
      const fetch = require('node-fetch');
      this.global.fetch = fetch;
    }
  }
};
