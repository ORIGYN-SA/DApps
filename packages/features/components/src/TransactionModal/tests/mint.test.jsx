import { render, screen } from '../../../../../../testUtils';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

import { Transaction } from '..';


const sale_obj = {
  token: 'Token not defined',
  amount: 'AMOUNT'
};

const props = {
  modalData: {
    trans_index: 'IndexTrx',
    token_id: 'TokenId',
    type_txn: 'Mint',
    message: 'Mint',
    accounts: 'AccountsHere',
    principals: 'PrincipalsHere',
    mint_from: 'Mint from',
    mint_to: 'Mint to',
    sale: sale_obj
  },
};

describe('Component/TransactionModal/Mint', () => {
  it('should display MINT transaction values', () => {
    const { getByText } = render(Transaction(props));
    screen.getByText(props.modalData.mint_to);
    screen.getByText(props.modalData.mint_from);
    screen.getByText(props.modalData.type_txn);
    screen.getByText(props.modalData.sale.amount);
    screen.getByText(props.modalData.sale.token);
  });
});
