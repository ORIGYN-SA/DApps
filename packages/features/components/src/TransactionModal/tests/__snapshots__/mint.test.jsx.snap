// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/TransactionModal/Mint should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiBox-root css-g5ybiu"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
        >
          Transaction type:
        </p>
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
        >
          Mint
        </h5>
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-6 css-1osj8n2-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
            >
              Mint from:
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
            >
              Mint from
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-6 css-1osj8n2-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
            >
              Mint to:
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
            >
              Mint to
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-hpgf8j"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-6 css-1osj8n2-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
            >
              Token:
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1rzlxhl-MuiTypography-root"
            >
              Token not defined
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-6 css-1osj8n2-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
            >
              Amount:
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1rzlxhl-MuiTypography-root"
            >
              AMOUNT
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
